Fix NullPointerException cause game crashed

Wulian233 opened the pull request:
### Fixes #14

Fix the NullPointerException error caused by `stitchResults` being null during game initialization.

* Add a null check for `stitchResults` before attempting to read its length in the `loadCachedData` method.
* Return an empty `ConcurrentHashMap` if `stitchResults` is null in the `loadCachedData` method.

### Changes

The mod description supports localization

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/steves-underwater-paradise/blinkload/pull/15?shareId=7b21f3c2-6c8c-478a-999f-ba14c676aadc).

---

Edit: Well, it seems like I fixed more than just a bug, I also added a new smalle feature and upgraded the mod version to 1.2

Wulian233 committed:
Fix NullPointerException cause game crashed

Fixes #14

Fix the NullPointerException error caused by `stitchResults` being null during game initialization.

* Add a null check for `stitchResults` before attempting to read its length in the `loadCachedData` method.
* Return an empty `ConcurrentHashMap` if `stitchResults` is null in the `loadCachedData` method.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/steves-underwater-paradise/blinkload/issues/14?shareId=XXXX-XXXX-XXXX-XXXX).

Wulian233 commented:
@Steveplays28 I tried to fix this crashing error with Copilot, could you review if this is the correct fix? Thank you! I don't know if the other two crashing errors are also caused by this

https://github.com/steves-underwater-paradise/blinkload/issues/13
https://github.com/steves-underwater-paradise/blinkload/issues/12

Wulian233 commented on common/src/main/java/io/github/steveplays28/blinkload/client/cache/BlinkLoadCache.java:
```diff
@@ -105,6 +105,9 @@ public static void cacheData(@NotNull StitchResult stitchResult) {
 			// Convert the JSON data to a Java object
 			@NotNull var stitchResults = JsonUtil.getGson().fromJson(reader, StitchResult[].class);
+			if (stitchResults == null) {
+				return new ConcurrentHashMap<>();
```
I'm not sure whether should `return cachedData` here. I also used Copilot on https://github.com/steves-underwater-paradise/blinkload/issues/11, and got the same fix as the issue above, but with a different return statement. Maybe we can solve all 4 issues at the same time.

https://github.com/steves-underwater-paradise/blinkload/issues/11

Steveplays28 commented:
Thanks @Wulian233, I'll do a review in the weekend. I don't have time currently

Wulian233 committed:
The mod description supports localization

Wulian233 committed:
upgrade gradle

Steveplays28 commented:
@Wulian233 If you update Gradle to `v8.11.1` you also need to update Architectury Loom to `v1.9-SNAPSHOT`

Wulian233 committed:
loom 1.9

Wulian233 commented:
Sorry I forgot. Upgrading many dependencies can always be less than pleasant🙂, I hope this won't break anything

Steveplays28 commented:
No worries, should be ok. I've tested that version combination already

sherpascott commented:
I was having this same issue, but the issue stopped happening recently. I didn't update the mod or anything, so it must be a mod conflict. I'm now saving about 15 seconds on modpack loading times!

Steveplays28 commented:
@sherpascott I'm glad to hear that the mod helps!
I'll get this PR merged as soon as possible to fix all open issues on these crashes.

Wulian233 commented:
Note: Date `## 'v1.2.0' - 14/12/2024` in `changelog.md`
Need to change again at the time of merge (kind reminder PR review)

Steveplays28 committed:
fix: :bug: Catch `JsonSyntaxException` and log error instead of crashing on `IOException`

The cache is now re-created when an exception is thrown while loading the cache, instead of throwing the exception.

Steveplays28 committed:
docs: :memo: Update changelog

Steveplays28 commented:
I forgot to run the CI, seems like the wrapper validation is failing.

Steveplays28 commented:
I've fixed it in #17.
